src/PrestaShopBundle/Form/Admin/Sell/Product/RedirectOptionType.php

```diff
@@ -145,16 +145,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             $form = $event->getForm();
             $targetField = $form->get('target');
             $targetOptions = $targetField->getConfig()->getOptions();
-            switch ($data['type']) {
-                case RedirectType::TYPE_CATEGORY_PERMANENT:
-                case RedirectType::TYPE_CATEGORY_TEMPORARY:
-                    $dataEntity = 'category';
-                    break;
-                case RedirectType::TYPE_PRODUCT_PERMANENT:
-                case RedirectType::TYPE_PRODUCT_TEMPORARY:
-                default:
-                    $dataEntity = 'product';
-                    break;
+            $dataEntity = 'product';
```

Comment:
```suggestion
            $dataType = $data['type'] ?? RedirectType::TYPE_PRODUCT_PERMANENT;
            switch ($dataType)
```

This is shorter and avoid the if block